packages/web/src/js/Dropdown.ts

```diff
@@ -0,0 +1,203 @@
+  }
+
+  static get NAME() {
+    return NAME;
+  }
+
+  getOptions() {
+    const options = { ...this.options };
+    const dataset = this.element?.dataset;
+    const optionsPlacement = dataset?.placement;
+    const optionsOffset = dataset?.offset;
+    const optionsPadding = dataset?.padding;
+    const optionsAutoClose = dataset?.autoclose;
+    if (optionsPlacement) options.placement = optionsPlacement;
+    if (optionsOffset) options.offset = Number(optionsOffset);
+    if (optionsPadding) options.padding = Number(optionsPadding);
+    if (optionsAutoClose) options.autoClose = Boolean(!optionsAutoClose);
+
+    return options;
+  }
+
+  findAnchorElement() {
+    let anchor = this.element;
+    const { reference } = this.element.dataset;
+    if (
+      reference &&
+      reference === 'parent' &&
+      this.element.parentElement &&
+      this.element.parentElement.classList.contains(CLASSNAME_DROPDOWN)
+    ) {
+      anchor = this.element.parentElement;
+    }
+
+    return anchor;
+  }
+
+  findTargetElement() {
+    let target = null;
+    if (this.element.dataset.target) {
+      target = SelectorEngine.findOne(`#${this.element.dataset.target}`);
+    }
+
+    return target;
+  }
+
+  updateTriggerElement(open: boolean = this.state.open) {
+    this.element.classList.toggle(CLASSNAME_EXPANDED, open);
+    this.element.setAttribute(ARIA_EXPANDED_ATTRIBUTE, open);
+    this.element.setAttribute(ARIA_CONTROLS_ATTRIBUTE, this.element.dataset.target);
+  }
+
+  updateTargetElement(open: boolean = this.state.open) {
+    this.target?.classList.toggle(CLASSNAME_OPEN, open);
+  }
+
+  alignTargetElement() {
+    const options = this.getOptions();
+    const { placement } = options;
+    const middleware = [
+      offset(options.offset),
+      flip(),
+      shift({
+        padding: options.padding,
+      }),
+    ];
+    if (this.anchor && this.target) {
+      computePosition(this.anchor, this.target, { placement, middleware }).then(({ x, y }) => {
+        this.target &&
+          Object.assign(this.target.style, {
+            left: `${x}px`,
+            top: `${y}px`,
+          });
+      });
+    }
+  }
+
+  cleanUp() {
+    if (this.anchor && this.target) {
+      autoUpdate(this.anchor, this.target, this.alignTargetElement.bind(this), this.autoUpdateOptions);
+    }
+  }
+
+  createAutoCloseEvent() {
+    const closeHandler = (event: Event) => {
+      const shouldClose = clickOutsideElement(this.target as HTMLElement, event);
+      if (event.target && shouldClose) {
+        this.hide();
+        document.removeEventListener('click', closeHandler);
+        // EventHandler.on(document, 'click', closeHandler); // Not working as expected, why?
+      }
+    };
+    document.addEventListener('click', closeHandler);
+    // EventHandler.off(document, 'click', closeHandler); // Not working as expected, why?
+  }
+
+  show() {
+    this.state.open = true;
+    EventHandler.trigger(this.target as HTMLElement, EVENT_SHOW);
```

Comment:
I think that `as HTMLElement` is an antipattern in this case. There is no need for this in any other classes we have so the problem will be somewhere else.